### ISSUE TITLE
Black screen when using colmap GUI via docker

### ISSUE THREAD

**Issue by mjlbach** [colmap/colmap#1808]
**Describe the bug**
When I open the colmap gui from within a docker container, the gui is black. This is resolved by setting QT_XCB_GL_INTEGRATION=xcb_egl

This issue is part PSA (how to run colmap's gui via docker) and part issue report, as it would be good to resolve/document this for others.

**To Reproduce**

```
docker run \
    -e DISPLAY=:0 \
    -v /tmp/.X11-unix:/tmp/.X11-unix \
    --gpus all \
    --privileged \
    -it colmap colmap gui
```

To fix:
```
docker run \
    -e QT_XCB_GL_INTEGRATION=xcb_egl \
    -e DISPLAY=:0 \
    -v /tmp/.X11-unix:/tmp/.X11-unix \
    --gpus all \
    --privileged \
    -it colmap colmap gui
```

**Expected behavior**
I would expect the container by default would support X11 forwarding, i'm not sure if this is a QT issue that can be worked around in the build process, or if the environmental variable should be added to the container image.

**Screenshots**
If applicable, add screenshots to help explain your problem. Input images and/or output reconstructions are usually helpful.
Without QT_XCB_GL_INTEGRATION=xcb_egl :
<img width="1015" alt="image" src="https://user-images.githubusercontent.com/13316262/219968906-2d1b04f7-f72a-4bdd-a768-387947580d52.png">

With QT_XCB_GL_INTEGRATION=xcb_egl 
<img width="1024" alt="image" src="https://user-images.githubusercontent.com/13316262/219968879-b7534bd2-a2f6-4064-8285-67e69644d0b0.png">

**Environment:**
 - OS: Fedora 37 (docker)
 - COLMAP Version: 3.8 (public docker container)
 - Capture Device: N/A

**Comment by ahojnnes** [comment]
Thanks for reporting this issue. This is probably something that could either be documented in the docker/readme or added to a quick-start-gui.sh script or similar?

**Comment by changh95** [comment]
This worked for me! Thanks a lot!

**Comment by ahojnnes** [comment]
Thanks @changh95, for the CUDA fix, did you change the versions in the docker container image or the host OS/CUDA versions?

**Comment by changh95** [comment]
> Yes - below are my settings. My host machine is Ubuntu 22.04 with RTX 3080. 

Dockerfile:
```
#FROM nvidia/cuda:11.7.0-devel-ubuntu22.04
#CUDA 11.7.0 + ubuntu22.04 does not work. Use 11.6.0 + ubuntu20.04 instead
FROM nvidia/cuda:11.6.0-devel-ubuntu20.04

ARG COLMAP_VERSION=dev
ARG CUDA_ARCHITECTURES=native

... (Same as master)
```

Docker command:
```
docker run \
    -e QT_XCB_GL_INTEGRATION=xcb_egl \
    -e DISPLAY=$DISPLAY \
    -v /tmp/.X11-unix:/tmp/.X11-unix \
    -v /home/changh95/data:/data \
    --env="QT_X11_NO_MITSHM=1" \
    --env="XAUTHORITY=$XAUTH" \
    --volume="$XAUTH=$XAUTH" \
    --network host \
    --gpus all \
    --privileged \
    --runtime=nvidia \
    -it colmap:latest colmap gui
```

**Comment by changh95** [comment]
> Thanks @changh95, for the CUDA fix, did you change the versions in the docker container image or the host OS/CUDA versions?

Yes - below are my settings. My host machine is Ubuntu 22.04 with RTX 3080. 

Dockerfile:
```
#FROM nvidia/cuda:11.7.0-devel-ubuntu22.04
#CUDA 11.7.0 + ubuntu22.04 does not work. Use 11.6.0 + ubuntu20.04 instead
FROM nvidia/cuda:11.6.0-devel-ubuntu20.04

ARG COLMAP_VERSION=dev
ARG CUDA_ARCHITECTURES=native

... (Same as master)
```

Docker command:
```
docker run \
    -e QT_XCB_GL_INTEGRATION=xcb_egl \
    -e DISPLAY=$DISPLAY \
    -v /tmp/.X11-unix:/tmp/.X11-unix \
    -v /home/changh95/data:/data \
    --env="QT_X11_NO_MITSHM=1" \
    --env="XAUTHORITY=$XAUTH" \
    --volume="$XAUTH=$XAUTH" \
    --network host \
    --gpus all \
    --privileged \
    --runtime=nvidia \
    -it colmap:latest colmap gui
```

**Comment by ahojnnes** [comment]
I believe this has been fixed.
